src/dash/issues/DashIssuesPage.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useIssueList, useCreateIssue, useLatestIssue } from "../../api/hooks";
+import { Table, Button, Modal, Form, Spinner } from "react-bootstrap";
+import { Link, useHistory } from "react-router-dom";
+import { useForm, ErrorMessage } from "react-hook-form";
+
+import "./DashIssuesPage.scss";
+import Field from "../../shared/form/Field";
+import ERRORS from "../../shared/errors";
+import { Issue } from "../../shared/types";
+
+const ISSUES_PER_VOLUME = 6;
+
+interface IssueCreateModalProps {
+  show: boolean;
+  setShow: (show: boolean) => void;
+  latestIssue: Issue;
+}
+
+interface IssueCreateFormVals {
+  issueNum: number;
+  volumeNum: number;
+}
+
+const IssueCreateModal: React.FC<IssueCreateModalProps> = (
+  props: IssueCreateModalProps
+) => {
+  const { handleSubmit, register, errors } = useForm<IssueCreateFormVals>({
+    defaultValues: {
+      volumeNum:
+        props.latestIssue.issue_num <= ISSUES_PER_VOLUME
+          ? props.latestIssue.volume_num + 1
+          : props.latestIssue.volume_num,
+      issueNum: (props.latestIssue.issue_num + 1) % ISSUES_PER_VOLUME,
+    },
+  });
+
+  const history = useHistory();
+
+  const [createIssue, ,] = useCreateIssue();
+
+  const [nonFieldErrors, setNonFieldErrors] = useState<string[]>([]);
+
+  const closeModal = () => {
+    setNonFieldErrors([]);
+    props.setShow(false);
+  };
+
+  const onSubmit = async (vals: IssueCreateFormVals) => {
+    const resp = await createIssue({
+      issue_num: vals.issueNum,
+      volume_num: vals.volumeNum,
+    });
+    if (resp.success) {
+      setNonFieldErrors([]);
+      history.push(`issues/${resp.data.id}`);
+    } else {
+      setNonFieldErrors(resp.error.non_field_errors || []);
+    }
+  };
+
+  return (
+    <Modal show={props.show} onHide={closeModal}>
+      <Modal.Header closeButton>Create New Issue</Modal.Header>
+      <Modal.Body>
+        <Form.Label>Issue Name:</Form.Label>
+        <Form id="createIssueForm" onSubmit={handleSubmit(onSubmit)} inline>
+          <Form.Label>v</Form.Label>
+          <Field
+            name="volumeNum"
+            className="volume-num-input"
+            type="text"
+            maxLength={3}
+            ref={register({
+              required: true,
+              pattern: /[0-9]{3}/,
+              min: 0,
+              maxLength: 3,
+            })}
```

Comment:
It would be simpler to make this field `type="number"` and set min/max attributes instead.

Reply:
The maxLength prevents you from typing in more than three numbers, but that doesn't work with `type="number"`, as far as I know.

Reply:
`maximum="999"` should work, I believe?

Reply:
The min/max properties only affect the spinner buttons that increment or decrement the value, they don't actually affect the text input at all.

Reply:
Huh, TIL. In that case the only thing I can think of is changing the pattern so that leading zeroes aren't allowed.

Reply:
I think I wasn't clear about the behaviour I wanted, I want the user to be able to only type in three characters at which point the text input will stop accepting keystrokes, and the pattern doesn't affect the actual input, only the browser validation afterwards. AFAIK, the only way to do this is the `maxLength` attribute.

Reply:
You're right on the `maxLength`. Just to prevent headaches later, I'd still suggest modifying the validation pattern to disallow leading zeros though.

This is for later, but it would be nice if focus could be transitioned to the issue text box once the third digit is entered (or `i` is pressed).